src/Highlighter.ts

```diff
@@ -10,7 +10,7 @@ interface IOptions {
   snapMathJax?: boolean;
   snapWords?: boolean;
   className?: string;
-  onClick?: (highlight?: Highlight) => void;
+  onClick?: (highlight?: Highlight | null) => void;
```

Comment:
Does this method need to handle all 3 cases? `onClick(highlight)`, `onClick()`, and `onClick(null)`? It seems a little confusing. From the references to `onClick` in this file it does not seem like `highlight` would ever be null... when would that occur?

Also, if `highlight` could be null, would it be worth creating an Optional type for readability? Example:

```ts
export type Optional<T> = T | null | undefined
```